### ISSUE TITLE
Issues with UpdatePropertyTest.cs

### ISSUE THREAD

**Issue by mikepizzo** [OData/WebApi#2696]
The update property test unit tests seem to have a couple of issues:
1) [CanUpdateComplexProperty](https://github.com/xuzhg/WebApiSample/blob/main/Microsoft.AspNet.OData.Test/CRUD/UpdatePropertyTest.cs#:~:text=const%20string%20payload%20%3D%20%22,Patch%22) seems to have an unnecessary value wrapper around the complex property.

2) We should support PUT in addition to PATCH to a primitive property (CanUpdatePrimitiveProperty and Can UpdateEnumProperty)

### Assemblies affected

WebAPI 7.x

### Additional detail

Not sure which of these may represent bugs/feature gaps in the underlying code; the first one is concerning.

**Comment by xuzhg** [comment]
@mikepizzo 

<h3>For the first issue:</h3>

The project is using OData lib 6.x version and Web API OData 5.9.1. In those versions, "value" is required.

Otherwise, (for example, I removed the "value"), I get the following error message.
`
"A top-level property with name 'Street' was found in the payload; however, property and collection payloads must always have a top-level property with name 'value'."
`

<h3>For the second issue:</h3>

I added the "put" for those tests, it works fine. See at: https://github.com/xuzhg/WebApiSample/commit/5afe46f5b4c5fac217c615f87493171e451b07e8
